src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -567,6 +591,34 @@ private async Task<List<string>> ExecuteCommandAsync(IExecutionContext context,
         }
 #endif
 
+#if OS_LINUX
+        private async Task<bool> IsMobyInstalled(IExecutionContext context)
+        {            
+            try
+            {
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    await processInvoker.ExecuteAsync(
+                                            workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Work),
+                                            fileName: "dpkg",
```

Comment:
what if `dpkg` not exists on private linux agent? is there any other way to detect moby?